routes/validators.go

```diff
@@ -210,3 +225,64 @@ func (fes *APIServer) UnregisterAsValidator(ww http.ResponseWriter, req *http.Re
 		return
 	}
 }
+
+func (fes *APIServer) GetValidatorByPublicKeyBase58Check(ww http.ResponseWriter, req *http.Request) {
+	// Parse ValidatorPublicKeyBase58Check from URL.
+	vars := mux.Vars(req)
+	validatorPublicKeyBase58Check, exists := vars["publicKeyBase58Check"]
+	if !exists {
+		_AddBadRequestError(ww, "GetValidatorByPublicKeyBase58Check: must provide a ValidatorPublicKeyBase58Check")
+		return
+	}
+
+	// Create UTXO view.
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddInternalServerError(ww, "GetValidatorByPublicKeyBase58Check: problem getting UTXO view")
+		return
+	}
+
+	// Convert ValidatorPublicKeyBase58Check to ValidatorPKID.
+	validatorPKID, err := fes.getPKIDFromPublicKeyBase58Check(utxoView, validatorPublicKeyBase58Check)
+	if err != nil || validatorPKID == nil {
+		_AddInternalServerError(ww, "GetValidatorByPublicKeyBase58Check: problem retrieving validator PKID")
+		return
+	}
+
+	// Get validator by PKID.
+	validatorEntry, err := utxoView.GetValidatorByPKID(validatorPKID)
+	if err != nil {
+		_AddInternalServerError(ww, "GetValidatorByPublicKeyBase58Check: problem retrieving validator")
+		return
+	}
+	if validatorEntry == nil {
+		_AddNotFoundError(ww, "GetValidatorByPublicKeyBase58Check: validator not found")
+		return
+	}
+
+	// Convert Domains [][]byte to []string.
+	var domains []string
+	for _, domain := range validatorEntry.Domains {
+		domains = append(domains, string(domain))
+	}
+
+	// Convert ValidatorEntry to ValidatorResponse.
+	validatorResponse := &ValidatorResponse{
```

Comment:
let's move the conversion of the core ValidatorEntry type to the backend ValidatorResponse type to a separate function in case we need to re-use it.